test/functional/rpc_getblockstats.py

```diff
@@ -70,90 +86,63 @@ def run_test(self):
         self.sync_all()
         node.generate(1)
 
-        start_height = 101
-        max_stat_pos = 2
-        stats = [node.getblockstats(height=start_height + i) for i in range(max_stat_pos+1)]
+        self.expected_stats = self.get_stats()
+
+        blocks = []
+        tip = node.getbestblockhash()
+        blkhash = None
+        height = 0
+        while tip != blkhash:
+            blkhash = node.getblockhash(height)
+            blocks.append(node.getblock(blkhash, 0))
+            height += 1
+
+        with open(filename, "w") as f:
+            f.write("""{\n  "stats": """)
```

Comment:
commit: Tests: Save and load block and corresponding expected statistics

Instead of hand-crafting the JSON, I think it's better to just create a dict with `stats` and `blocks` keys and json.dump that. It seems you might be doing this for better control over the indentation, but the data file added in the next commit has each stat on its own line.

Reply:
It puts the stats on separate lines, but all the blocks on a single line. Seemed like a good idea at the time (multiline stats hand editable, block data as a minimal blob), but doesn't make much difference.